reactioncommerce/reaction
# Order data fetch

Resolves #2837 #2952 #2768

## TEST #2837 
### Expected behavior

When there's a filter applied to the orders table and `select all` action performed, only the filtered orders should be selected.

### Actual Behavior
Even with a filter applied, `select all` selects all orders present in the table.

### How to test
- Have a couple of orders in different states/stages
- Visit the orders dashboard (click on the orders icon in the sidebar)
- Apply any kind of filter to the orders e.g.  filter by status
- Click Select All checkbox on order table header
- Change filters observe that only the filtered orders are selected

## TEST #2952
### Expected behavior

Orders table should have the same number of rows as orders.
For example: If there are two orders, have just two rows displayed.

### Actual Behavior
Extra empty rows appear even when there are orders.

### How to test
- Create two.more orders
- Open order dashboard, observe that table grows/shrinks depending on the number of orders
- If there are no results the default 3 rows show up


## TEST #2768 
### Problem
Order Dashboard Table was using ReactTable's way of passing publication/collection as opposed to passing it the actual data

### How to test 
- Look through code, see that Order Table passes its own data to the React Table implementation.
